Automattic/jetpack
# Protect: React Query

## Proposed changes:

* Adds React Query
* Removes Redux

### Other information:


## Jetpack product discussion



https://github.com/Automattic/jetpack-scan-team/issues/1382

## Does this pull request change what data or activity we track or use?



No

## Testing instructions:






Note: using your local development environment to test this PR will allow you to view the React Query developer tools via a floating icon on the bottom right of the screen. 

Monitor the network tab for the duration of your testing. Ensure no async requests are triggered on page load (data comes from initial state), request polling occurs for in-progress scans and fixers, and related data is refreshed after actions are made (i.e. refresh scan status after ignoring a threat).

* Test _all the things_:
  * Install on a brand new site, activate the free plan.
  * Review your initial Protect report.
  * Install a vulnerable plugin or two, review your second report (re-trigger a second free report from Jetpack Debug).
  * Activate the firewall, configure and test brute force protection and the IP lists.
  * Upgrade your plan.
  * Review the new scan report.
  * Add more threats to your site with the Jetpack Debug Helper plugin.
  * Manually start a new scan.
  * Ignore some threats, verify they are moved to history.
  * Auto-fix some threats, add credentials along the way.
  * Activate and test automatic firewall rules (i.e. ensure a request including `?foo=<script>` in the URL is blocked when automatic rules are enabled).
